opsdroid/connector/matrix/connector.py

```diff
@@ -223,19 +246,27 @@ async def get_nick(self, roomid, mxid):
         setting.
         """
         if self.room_specific_nicks:
-            try:
-                return await self.connection.get_room_displayname(roomid, mxid)
-            except Exception:  # pylint: disable=W0703
-                # Fallback to the non-room specific one
-                logging.exception("Failed to lookup room specific nick for %s.", mxid)
+            if roomid is None:
+                return mxid
 
-        try:
-            return await self.connection.get_display_name(mxid)
-        except MatrixRequestError as mre:
-            # Log the error if it's not the 404 from the user not having a nick
-            if mre.code != 404:
-                logging.exception("Failed to lookup nick for %s.", mxid)
+            res = await self.connection.joined_members(roomid)
+            if isinstance(res, nio.JoinedMembersError):
+                logging.exception("Failed to lookup room members for %s.", roomid)
+                return mxid
+
+            for member in res.members:
+                if member.user_id == mxid:
+                    return member.display_name
+
+            return mxid
+
+        res = await self.connection.get_displayname(mxid)
+        if isinstance(res, nio.ProfileGetDisplayNameError):
+            logging.exception("Failed to lookup nick for %s.", mxid)
```

Comment:
I think that `.exception` uses the current stacktrace when inside an `except:` block. I am not sure how you should pass the exception object to it in this case.

Also I think you need to use `_LOGGER` not `logging` to make sure the logging line knows the calling scope.